fix(block): make `inner` aware of title positions

jan-ferdinand opened the pull request:
Previously, when computing the inner rendering area of a block, all titles were assumed to be positioned at the top. If some or all titles were at the bottom, this caused various incorrect computations of the inner area, depending on which borders were set; no border at the bottom would lead to an area that's too big, no border at the top but at the bottom and _all_ titles at the bottom would lead to an area that's too small.

Now, when computing the inner area, the positions of the titles are taken into account.

fix #658

orhun reviewed (commented):
Can you also add a test case where you cover the issue in #658? (you can assert the contents of the buffer with `assert_buffer`)

jan-ferdinand commented:
> add a test case where you cover the issue in #658

Good suggestion. I have amended the PR accordingly. Let me know if anything can be improved. :relaxed:

joshka commented on src/widgets/block.rs:
```diff
@@ -532,6 +533,16 @@ impl<'a> Block<'a> {
     }
 
+    fn have_title_at_position(&self, position: Position) -> bool {
+        if self.titles.iter().any(|t| t.position == Some(position)) {
+            return true;
+        }
+        if self.titles.iter().any(|t| t.position.is_none()) {
+            return self.titles_position == position;
+        }
+        false
```
slightly simpler (nit):

```suggestion
        self.titles
            .iter()
            .any(|title|
                title.position == Some(position)
                || title.position.is_none() && self.titles_position == position)
```

jan-ferdinand replied:
How about:

``` rust
        self.titles.iter().any(|title| match title.position {
            Some(title_position) => title_position == position,
            None => self.titles_position == position,
        })
```

joshka replied:
Good idea, but you could go one step futher still: :)
```rust
self.titles.iter.any(|title| position == title_position.unwrap_or(self.titles_position))
```

joshka reviewed (approved):
Can you please suggest a simple and concise explanation of the bug?
If you're able, add it to the body of a commit message to make it automatically fill the changelog. If not, we can add the wording when squashing the commit.

jan-ferdinand commented:
I have added an explanation of the bug and the fix to both the description of this PR and the commit message. Let me know should brevity be lacking, or if the level of detail could be improved upon (in either direction).

Also, I have reworked the part that @joshka had a nit about, without accepting the suggestion. If you prefer to have your suggestion exactly, I'll incorporate that instead.

jan-ferdinand commented:
Thanks for the shortened bug description as well as the improved code snippet. :+1: I've incorporated both and amended the PR accordingly.

orhun commented:
Thanks for your contribution!
